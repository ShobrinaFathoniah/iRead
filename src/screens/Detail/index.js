import {
  View,
  Share,
  ScrollView,
  TouchableOpacity,
  RefreshControl,
  Image,
} from 'react-native';
import React, {useEffect} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {getDataDetail} from './redux/action';
import {
  CircleButton,
  Courgette,
  LibreBaskerville,
  LoadingBar,
  NoConnection,
  Rating,
} from '../../components';
import styles from './style';
import {moderateScale} from 'react-native-size-matters';
import {LIGHT_BLUE_300, LIGHT_BLUE_600, RED_500} from '../../helpers/colors';
import {setHeart, setRefreshing} from '../../store/globalAction';
import {numberToIDR} from '../../helpers/numberToIDR';
import {Notification} from '../../helpers/notification';

const Detail = ({route, navigation}) => {
  const {params} = route.params;
  const id_book = params.idBook;

  const dispatch = useDispatch();
  const {dataToken} = useSelector(state => state.login);
  const {detail} = useSelector(state => state.detailBook);
  const {heart, idBook} = useSelector(state => state.global);

  const {refreshing, isLoading, connection} = useSelector(
    state => state.global,
  );

  const getDetail = () => {
    dispatch(getDataDetail(dataToken, id_book));
  };

  useEffect(() => {
    getDetail();
  }, []);

  const shareData = () => {
    Share.share({
      message: `Saya ingin merekomendasikan Anda Buku yang berjudul  '${
        detail.title
      }', buku tersebut merupakan karangan dari ${
        detail.author
      }, memiliki jumlah halaman ${
        detail.page_count
      } halaman. Buku ini memiliki harga yaitu sebesar ${numberToIDR(
        detail.price,
      )}. Ayo buruan beli! Stok buku yang tersedia berjumlah ${
        detail.stock_available
      } buku.`,
    });
  };

  const lovedBook = () => {
    Notification.configure();
    Notification.createChannel('1');

    if (heart && idBook) {
      dispatch(setHeart({heart: false, idBook: ''}));

      Notification.sendNotifictaion(
        '1',
        'Unloved Book!',
        `Kamu Batal menyukai Buku ${detail.title}!`,
      );
    } else {
      dispatch(setHeart({heart: true, idBook: id_book}));

      Notification.sendNotifictaion(
        '1',
        'Loved Book!',
        `Kamu menyukai Buku ${detail.title}!`,
      );
    }
  };

  const onRefresh = () => {
    dispatch(setRefreshing(true));
    getDetail();
  };

  const stock = detail.stock_available;
  const stockChecker = stock => (stock > 0 ? LIGHT_BLUE_600 : RED_500);

  const buttonBuy = stock => {
    if (stock > 0) {
      return (
        <View style={styles.containerButtonBuy}>
          <TouchableOpacity style={styles.buttonBuy}>
            <LibreBaskerville style={[styles.text, styles.textButtonBuy]}>
              Buy {numberToIDR(detail.price)}
            </LibreBaskerville>
          </TouchableOpacity>
        </View>
      );
    } else {
      return (
        <View style={styles.containerButtonBuy}>
          <View
            style={[
              styles.buttonBuy,
              {
                backgroundColor: RED_500,
              },
            ]}>
            <LibreBaskerville style={[styles.text, styles.textButtonBuy]}>
              Out of Stock
            </LibreBaskerville>
          </View>
        </View>
      );
    }
  };

  const detailScreen = () => {
    return (
      <View>
        <View style={styles.containerAllBookInfo}>
          <View style={styles.containerImage}>
            <Image
              style={styles.coverImage}
              source={{uri: detail.cover_image}}
            />
          </View>
          <View style={styles.containerBookInfo}>
            <LibreBaskerville
              type="Bold"
              style={[styles.text, styles.textTitle]}>
              {detail.title}
            </LibreBaskerville>

            <LibreBaskerville style={[styles.text, styles.textAuthor]}>
              {detail.author}
            </LibreBaskerville>

            <LibreBaskerville style={[styles.text, styles.textPublisher]}>
              {detail.publisher}
            </LibreBaskerville>
            <View style={styles.containerRating}>
              <Rating rating={detail.average_rating} />
              <LibreBaskerville style={[styles.text, styles.textRating]}>
                {detail.average_rating}
              </LibreBaskerville>

              <LibreBaskerville
                style={[styles.text, styles.textSold, {color: LIGHT_BLUE_300}]}>
                |
              </LibreBaskerville>

              <LibreBaskerville style={[styles.text, styles.textSold]}>
                {detail.total_sale}
              </LibreBaskerville>
              <LibreBaskerville style={[styles.text, styles.textSold]}>
                Sold
              </LibreBaskerville>
            </View>
          </View>
        </View>

        <View style={styles.containerInfoAvailableBooks}>
          <View style={styles.containerPages}>
            <LibreBaskerville
              type="Bold"
              style={[styles.text, styles.textPages]}>
              {detail.page_count}
            </LibreBaskerville>
            <LibreBaskerville style={[styles.text, styles.title]}>
              Pages
            </LibreBaskerville>
          </View>
          <View style={styles.containerStocks}>
            <LibreBaskerville
              type="Bold"
              style={[
                styles.text,
                styles.textStocks,
                {
                  color: stockChecker(stock),
                },
              ]}>
              {stock}
            </LibreBaskerville>
            <LibreBaskerville style={[styles.text, styles.title]}>
              Stocks
            </LibreBaskerville>
          </View>
        </View>

        {buttonBuy(stock)}

        <View style={styles.containerOverview}>
          <LibreBaskerville style={[styles.text, styles.titleSynopsis]}>
            Synopsis
          </LibreBaskerville>
          <Courgette style={[styles.text, styles.textSynopsis]}>
            {detail.synopsis}
          </Courgette>
        </View>
      </View>
    );
  };

  return (
    <ScrollView
      refreshControl={
        <RefreshControl onRefresh={onRefresh} refreshing={refreshing} />
      }>
      <View style={styles.allButtons}>
        <View>
          <CircleButton
            nameIcon="arrow-left"
            onPress={() => navigation.goBack()}
          />
        </View>
        <View style={styles.miniButtons2}>
          <CircleButton
            color={RED_500}
            style={{marginEnd: moderateScale(10)}}
            nameIcon={heart && idBook === id_book ? 'heart' : 'heart-o'}
            onPress={lovedBook}
          />
          <CircleButton nameIcon="share" onPress={shareData} />
        </View>
      </View>

      {LoadingBar(isLoading)}
      {connection ? detailScreen() : NoConnection(connection)}
    </ScrollView>
  );
};

export default Detail;
