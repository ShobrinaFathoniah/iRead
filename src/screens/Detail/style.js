import {StyleSheet} from 'react-native';
import {
  heightPercentageToDP,
  widthPercentageToDP,
} from 'react-native-responsive-screen';
import {moderateScale} from 'react-native-size-matters';
import {
  BLACK,
  DARK_PURPLE_100_trans,
  DARK_PURPLE_500,
  LIGHT_BLUE_300,
} from '../../helpers/colors';

const styles = StyleSheet.create({
  allButtons: {
    flexDirection: 'row',
    margin: moderateScale(10),
  },
  miniButtons2: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'flex-end',
  },
  miniButtonPdf: {
    margin: moderateScale(10),
  },
  text: {
    color: DARK_PURPLE_500,
    fontSize: moderateScale(12),
    letterSpacing: moderateScale(0.7),
  },
  textTitle: {
    fontSize: moderateScale(17),
    color: DARK_PURPLE_500,
    paddingRight: moderateScale(5),
  },
  coverImage: {
    height: heightPercentageToDP(20),
    width: widthPercentageToDP(33),
    resizeMode: 'contain',
    borderRadius: moderateScale(5),
  },
  containerAllBookInfo: {
    flexDirection: 'row',
    marginTop: moderateScale(5),
    paddingVertical: moderateScale(10),
  },
  containerImage: {
    flex: 1,
  },
  containerBookInfo: {
    flex: 2,
    margin: moderateScale(10),
  },
  textAuthor: {
    marginTop: moderateScale(10),
  },
  textPublisher: {
    marginTop: moderateScale(3),
  },
  containerRating: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: moderateScale(5),
  },
  textRating: {
    fontSize: moderateScale(14),
  },
  containerInfoAvailableBooks: {
    margin: moderateScale(10),
    flexDirection: 'row',
    justifyContent: 'center',
  },
  containerPages: {
    marginRight: moderateScale(30),
    alignItems: 'center',
  },
  textPages: {
    fontSize: moderateScale(15),
    marginTop: moderateScale(5),
  },
  containerStocks: {
    alignItems: 'center',
  },
  title: {
    fontSize: moderateScale(15),
  },
  textStocks: {
    fontSize: moderateScale(15),
    marginTop: moderateScale(5),
  },
  containerSold: {
    alignItems: 'center',
  },
  textSold: {
    fontSize: moderateScale(14),
    marginStart: moderateScale(5),
  },
  containerButtonBuy: {
    marginRight: moderateScale(10),
  },
  buttonBuy: {
    marginStart: moderateScale(10),
    padding: moderateScale(10),
    backgroundColor: LIGHT_BLUE_300,
    width: moderateScale(160),
    alignItems: 'center',
    borderRadius: moderateScale(5),
  },
  textButtonBuy: {
    fontSize: moderateScale(16),
    textAlign: 'center',
    color: BLACK,
  },
  containerOverview: {
    marginHorizontal: moderateScale(15),
  },
  titleSynopsis: {
    fontSize: moderateScale(18),
    marginTop: moderateScale(10),
  },
  textSynopsis: {
    fontSize: moderateScale(14),
    textAlign: 'justify',
    letterSpacing: moderateScale(0.7),
    color: DARK_PURPLE_500,
  },
  synopsisContainer: {
    marginTop: moderateScale(5),
    marginEnd: moderateScale(5),
    borderRadius: moderateScale(10),
  },
  containerAllButton: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    margin: moderateScale(10),
  },
  containerPdf: {
    height: heightPercentageToDP(80),
    margin: moderateScale(10),
    backgroundColor: DARK_PURPLE_100_trans,
  },
  pdf: {
    height: heightPercentageToDP(80),
    margin: moderateScale(5),
  },
});

export default styles;
