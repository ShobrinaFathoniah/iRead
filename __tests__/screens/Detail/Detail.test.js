import * as React from 'react';
import Detail from '../../../src/screens/Detail';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../../src/helpers/reduxTesting';

test('Detail Snapshoot', () => {
  jest.useFakeTimers();

  const mockedParams = {
    route: {params: {params: '6231453513c01e6f8b566ecb'}},
    navigation: '',
  };

  const snap = create(ContainerTesting(<Detail {...mockedParams} />));
  expect(snap).toMatchSnapshot();
});
