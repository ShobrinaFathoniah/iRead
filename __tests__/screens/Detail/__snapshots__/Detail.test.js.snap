// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Snapshoot 1`] = `
<RCTScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 15.714285714285714,
        }
      }
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#53548799",
              "borderRadius": 157.14285714285714,
              "borderWidth": 1.5714285714285714,
              "height": 62.857142857142854,
              "justifyContent": "center",
              "opacity": 1,
              "width": 62.857142857142854,
            }
          }
          testID="circleButton"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#272b5a",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#53548799",
              "borderRadius": 157.14285714285714,
              "borderWidth": 1.5714285714285714,
              "height": 62.857142857142854,
              "justifyContent": "center",
              "marginEnd": 15.714285714285714,
              "opacity": 1,
              "width": 62.857142857142854,
            }
          }
          testID="circleButton"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#c4131e",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#53548799",
              "borderRadius": 157.14285714285714,
              "borderWidth": 1.5714285714285714,
              "height": 62.857142857142854,
              "justifyContent": "center",
              "opacity": 1,
              "width": 62.857142857142854,
            }
          }
          testID="circleButton"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#272b5a",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 7.857142857142857,
            "paddingVertical": 15.714285714285714,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "borderRadius": 7.857142857142857,
                "height": 267,
                "resizeMode": "contain",
                "width": 247.5,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 2,
              "margin": 15.714285714285714,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Bold",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 26.714285714285715,
                  "letterSpacing": 1.0999999999999999,
                  "paddingRight": 7.857142857142857,
                },
              ]
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 18.857142857142858,
                  "letterSpacing": 1.0999999999999999,
                  "marginTop": 15.714285714285714,
                },
              ]
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 18.857142857142858,
                  "letterSpacing": 1.0999999999999999,
                  "marginTop": 4.714285714285714,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 7.857142857142857,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginRight": 15.714285714285714,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#F1E1A6",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#F1E1A6",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#F1E1A6",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#F1E1A6",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#F1E1A6",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#272b5a",
                    "fontSize": 22,
                    "letterSpacing": 1.0999999999999999,
                  },
                ]
              }
            />
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#53548799",
                    "fontSize": 22,
                    "letterSpacing": 1.0999999999999999,
                    "marginStart": 7.857142857142857,
                  },
                ]
              }
            >
              |
            </Text>
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#272b5a",
                    "fontSize": 22,
                    "letterSpacing": 1.0999999999999999,
                    "marginStart": 7.857142857142857,
                  },
                ]
              }
            />
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#272b5a",
                    "fontSize": 22,
                    "letterSpacing": 1.0999999999999999,
                    "marginStart": 7.857142857142857,
                  },
                ]
              }
            >
              Sold
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 15.714285714285714,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginRight": 47.14285714285714,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Bold",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 23.57142857142857,
                  "letterSpacing": 1.0999999999999999,
                  "marginTop": 7.857142857142857,
                },
              ]
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 23.57142857142857,
                  "letterSpacing": 1.0999999999999999,
                },
              ]
            }
          >
            Pages
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Bold",
                },
                Object {
                  "color": "#c4131e",
                  "fontSize": 23.57142857142857,
                  "letterSpacing": 1.0999999999999999,
                  "marginTop": 7.857142857142857,
                },
              ]
            }
          />
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 23.57142857142857,
                  "letterSpacing": 1.0999999999999999,
                },
              ]
            }
          >
            Stocks
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "margin": 15.714285714285714,
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 15.714285714285714,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#39c2dd",
                  "borderRadius": 7.857142857142857,
                  "marginStart": 15.714285714285714,
                  "padding": 15.714285714285714,
                  "width": 251.42857142857142,
                },
                Object {
                  "backgroundColor": "#535487",
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 25.142857142857142,
                    "letterSpacing": 1.0999999999999999,
                    "textAlign": "center",
                  },
                ]
              }
            >
              No e-Book
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginRight": 15.714285714285714,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#39c2dd",
                  "borderRadius": 7.857142857142857,
                  "marginStart": 15.714285714285714,
                  "padding": 15.714285714285714,
                  "width": 251.42857142857142,
                },
                Object {
                  "backgroundColor": "#c4131e",
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "fontFamily": "LibreBaskerville-Regular",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 25.142857142857142,
                    "letterSpacing": 1.0999999999999999,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Out of Stock
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 23.57142857142857,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          style={
            Array [
              Object {
                "fontFamily": "LibreBaskerville-Bold",
              },
              Object {
                "color": "#272b5a",
                "fontSize": 28.285714285714285,
                "letterSpacing": 1.0999999999999999,
                "marginTop": 15.714285714285714,
              },
            ]
          }
        >
          Synopsis
        </Text>
        <View
          style={
            Object {
              "borderRadius": 15.714285714285714,
              "marginEnd": 7.857142857142857,
              "marginTop": 7.857142857142857,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "LibreBaskerville-Regular",
                },
                Object {
                  "color": "#272b5a",
                  "fontSize": 22,
                  "letterSpacing": 1.0999999999999999,
                  "textAlign": "justify",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
